pythonforandroid/recipe.py

```diff
@@ -342,16 +354,19 @@ def download(self):
             return
 
         url = self.versioned_url
-        ma = match(u'^(.+)#md5=([0-9a-f]{32})$', url)
-        if ma:                  # fragmented URL?
-            if self.md5sum:
-                raise ValueError(
-                    ('Received md5sum from both the {} recipe '
-                     'and its url').format(self.name))
-            url = ma.group(1)
-            expected_md5 = ma.group(2)
-        else:
-            expected_md5 = self.md5sum
+        expected_digests = {}
+        for alg in set(hashlib.algorithms_guaranteed) | set(('md5', 'sha512', 'blake2b')):
+            expected_digest = getattr(self, alg + 'sum') if hasattr(self, alg + 'sum') else None
```

Comment:
@fuzzyTew fyi you could shorten

```py
getattr(self, alg + 'sum') if hasattr(self, alg + 'sum') else None
```

to

```py
getattr(self, alg + 'sum', None)
```

Reply:
Thanks, I didn't see that.  If you have any interest / effort, I may have trouble getting to this for a while.

The biggest issue with my code is that I provided no way to set the digest length, so you can't do 256-bit blake2 which pypi uses.

Might be reasonable also to add a note regarding other security norms of the system not yet being as tight as the new algorithms.